### ISSUE TITLE
Jetpack Sites: Drafts and trashed posts count incorrect

### ISSUE THREAD

**Issue by davidakennedy** [Automattic/wp-calypso#9871]
This may be hard to reproduce. I've experienced this multiple times, but not all the time. Just yesterday, I think I've nailed down the proper steps:

## Steps to reproduce
1. Starting at URL: davidakennedy.com (Jetpack-powered site)
2. Started writing a new post with Calypso.
3. Saved that post.
4. Use Calypso on another device (for me it's the web version on another laptop)
5. The draft does not show up.
6. Visit the site using WP Admin, and check posts: the draft is there.
6. May be related: I also see a `1` for the Trashed posts. But nothing is there.

## What I expected

I expect to see my draft posts on all devices after saving them.

## What happened instead

I can't see my draft posts in Calypso after saving them, and then accessing Calypso on a different device.

## Browser / OS version

Mac OS X 10.12.1 / Chrome 54.0.2840.98 and Safari 10.

## Screenshot / Video

![screen shot 2016-12-07 at 10 44 22 am](https://cloud.githubusercontent.com/assets/1473618/20974538/98f1ef8a-bc6a-11e6-907f-c1ee66364373.png)
![screen shot 2016-12-07 at 10 46 29 am](https://cloud.githubusercontent.com/assets/1473618/20974539/98f1d70c-bc6a-11e6-9b4f-b6e36b2bc656.png)

## Context / Source

As I mentioned above, this may be hard to reproduce. I've seen it especially when starting on one device and moving to another. This often happens because I start a draft on a device when I have an idea, but will pick it up on another later. Found while #dogfooding.

**Comment by catehstn** [comment]
I think I repro'd this - I created two drafts, both of them showed up under "schedule", but the count showed just 1. I was also switching between wp-admin and Calypso. I *think* the only thing I did in WP-Admin was manage media.

Attaching screenshot. There is another draft post below that doesn't show on the screen.

<img width="846" alt="screen shot 2017-02-05 at 20 50 36" src="https://cloud.githubusercontent.com/assets/6810182/22660334/5eac5ce0-eca1-11e6-84ee-24490bb588e4.png">
